### ISSUE TITLE
Firefox error on signInWithCustomToken with empty referer: auth/requests-from-referer-<empty>-are-blocked

### ISSUE THREAD

**Issue by CharlieDigital** [firebase/firebase-js-sdk#8417]
### Operating System

Mac M1, macOS 14.5

### Browser Version

Firefox 129.0

### Firebase SDK Version

10.12.5

### Firebase SDK Product:

Auth

### Describe your project's tooling

Vue + Vite building a browser extension

### Describe the problem

When making a call to `signInWithCustomToken`, Firebase auth receives an HTTP 403 from the identity endpoint with the following error:

![image](https://github.com/user-attachments/assets/010d5c34-7a6b-42ae-9236-af9da03f8dca)

This only occurs in Firefox and works as expected on Chrome.

![Screenshot 2024-08-06 at 10 29 28 PM](https://github.com/user-attachments/assets/40a8b575-c80f-447a-8ba5-9bff524914a7)

### Steps and code to reproduce issue

I have the code published in the following extensions (with minimal obfuscation):

Chrome version of the extension where everything works: https://chromewebstore.google.com/detail/turasapp/lpfijfdbgohlblnadiokliolkkeeblpo

Firefox version: https://addons.mozilla.org/en-US/firefox/addon/turas-app/

There is no functional difference in the output code.

May be related to this issue: https://github.com/firebase/firebase-js-sdk/issues/7997

**Comment by cabello** [comment]
Facing similar issue with `await signInWithCustomToken(auth, customFirebaseToken)` on Windows 10 + Chrome 126 and 127.
We issue the token in the backend using a service account and then use `signInWithCustomToken` using the client side API that is configured with a browser key that is restricted for our sites.
I don't see the option to supply a referer in this case.

**Comment by dlarocque** [comment]
Hi @CharlieDigital, it looks like https://addons.mozilla.org/en-US/firefox/addon/turas-app/ does not exist. Do you have another version we can take a look at?

**Comment by CharlieDigital** [comment]
Yup; I took it offline for the time being, but I've attached the compiled source here (largely identical except for the manifest):

[turas-firefox-2024-09-14-16-39.zip](https://github.com/user-attachments/files/16996971/turas-firefox-2024-09-14-16-39.zip)

**Comment by DellaBitta** [comment]
Hi all, are you still experiencing this issue? If so, can you share a stand alone, minimal app that reproduces the problem? Thanks!

**Comment by CharlieDigital** [comment]
I'm still interested in solving this, but may not be able to get to testing this again later this week (and also requires waiting for Mozilla to approve my app).  I will try to update all packages and publish a new version of the extension this week and test again.

**Comment by CharlieDigital** [comment]
Problem still occurs after updating the `firebase` package to 11.7.3:

<img width="604" alt="Image" src="https://github.com/user-attachments/assets/aeda3786-5e89-4774-97e2-c1512ca893ce" />

Source is available here (not obfuscated): https://addons.mozilla.org/en-US/firefox/addon/turas-app/

Attached as a package: 

[turas-firefox-2025-05-18-00-19.zip](https://github.com/user-attachments/files/20260342/turas-firefox-2025-05-18-00-19.zip)
